src/parquet/arrow/arrow-reader-writer-test.cc

```diff
@@ -1074,22 +1075,20 @@ TEST(TestArrowWrite, CheckChunkSize) {
       Invalid, WriteTable(*table, ::arrow::default_memory_pool(), sink, chunk_size));
 }
 
-class TestNestedSchemaRead : public ::testing::Test {
+class TestNestedSchemaRead : public ::testing::TestWithParam<Repetition::type> {
  protected:
-  virtual void SetUp() {
-    // We are using parquet low-level file api to create the nested parquet
-    CreateNestedParquet();
-    InitReader(&reader_);
-  }
+  // make it *3 to make it easily divisible by 3
+  const int NUM_SIMPLE_TEST_ROWS = SMALL_SIZE * 3;
+  std::shared_ptr<::arrow::Int32Array> values_array_ = nullptr;
 
-  void InitReader(std::shared_ptr<FileReader>* out) {
+  void InitReader() {
     std::shared_ptr<Buffer> buffer = nested_parquet_->GetBuffer();
     std::unique_ptr<FileReader> reader;
```

Comment:
With your changes, we should directly use `reader_` here instead of adding another local variable.